class.c

```diff
@@ -1334,13 +1336,25 @@ rb_mod_ancestors(VALUE mod)
     return ary;
 }
 
+struct subclass_traverse_data
+{
+    VALUE *buffer;
+    long count;
+    long maxcount;
+};
+
 static void
-class_descendants_recursive(VALUE klass, VALUE ary)
+class_descendants_recursive(VALUE klass, VALUE v)
 {
+    struct subclass_traverse_data *data = (struct subclass_traverse_data *) v;
+
     if (BUILTIN_TYPE(klass) == T_CLASS && !FL_TEST(klass, FL_SINGLETON)) {
-        rb_ary_push(ary, klass);
+        if (data->buffer && (data->count < data->maxcount || data->maxcount == -1)) {
```

Comment:
In what condition  can `data->buffer && data->maxcount == -1` be true?

Reply:
Good point. Removed `data->maxcount == -1`